utils/vhost/tcphttptunnel.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 guylewin, guy@lewin.co.il
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vhost
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"time"
+)
+
+type TcpHttpTunnelMuxer struct {
+	*VhostMuxer
+}
+
+func NewTcpHttpTunnelMuxer(listener net.Listener, timeout time.Duration) (*TcpHttpTunnelMuxer, error) {
+	mux, err := NewVhostMuxer(listener, getHostFromHttpConnect, nil, sendHttpOk, timeout)
```

Comment:
The fourth params in `NewVhostMuxer` is `hostRewriteFunc`, it's tricky but ok now. Maybe add TODO comments to refactoring later.

Reply:
Yeah it's kind of a stretch. I thought about adding another parameter, just a callback for successful connections, instead of re-using this parameter. What do you think?

Reply:
I agree.

Reply:
I agree.